diff -urN gnome-netstatus-2.7.91.orig/configure.in gnome-netstatus-2.7.91/configure.in
--- gnome-netstatus-2.7.91.orig/configure.in	2004-08-17 19:26:34.116236120 +0200
+++ gnome-netstatus-2.7.91/configure.in	2004-08-17 19:26:49.272931952 +0200
@@ -23,7 +23,7 @@
 dnl make sure we keep ACLOCAL_FLAGS around for maintainer builds to work
 AC_SUBST(ACLOCAL_AMFLAGS, "$ACLOCAL_FLAGS")
 
-ALL_LINGUAS="ar az bg bn ca cs cy da de el en_CA en_GB es fi fr ga gu hi hr hu id it ja ko lt mn ms nl nn no pa pl pt pt_BR ro ru sk sq sr sr@Latn sv ta th tr vi uk wa zh_CN"
+ALL_LINGUAS="ar az bg bn ca cs cy da de el en_CA en_GB es fi fr ga gu hi hr hu id it ja ko lt mn ms nb nl nn pa pl pt pt_BR ro ru sk sq sr sr@Latn sv ta th tr vi uk wa zh_CN"
 AM_GLIB_GNU_GETTEXT
 
 GNOME_COMPILE_WARNINGS(yes)
