diff -urN gnome-netstatus-2.26.0/configure.in gnome-netstatus-2.26.0.new/configure.in
--- gnome-netstatus-2.26.0/configure.in	2009-03-08 06:44:58.000000000 +0100
+++ gnome-netstatus-2.26.0.new/configure.in	2009-03-12 10:55:22.000000000 +0100
@@ -46,7 +46,8 @@
 PKG_CHECK_MODULES(NETSTATUS,
 			     libglade-2.0 >= 2.0.0
 			     libpanelapplet-2.0 >= 2.0.0
-			     gtk+-2.0 >= 2.14.0)
+			     gtk+-2.0 >= 2.14.0
+			     libnotify)
 
 AC_CHECK_HEADERS(sys/sockio.h)
 
diff -urN gnome-netstatus-2.26.0/src/netstatus-applet.c gnome-netstatus-2.26.0.new/src/netstatus-applet.c
--- gnome-netstatus-2.26.0/src/netstatus-applet.c	2009-03-08 06:25:52.000000000 +0100
+++ gnome-netstatus-2.26.0.new/src/netstatus-applet.c	2009-03-12 10:57:12.000000000 +0100
@@ -464,6 +464,9 @@
   netstatus_iface_set_name (applet->priv->iface, iface_name);
   g_free (iface_name);
 
+  /* Interfaces notification timer */
+  g_timeout_add (1000, (GSourceFunc) netstatus_ifaces_monitor_start, NULL);
+
   key = panel_applet_gconf_get_full_key (PANEL_APPLET (applet), "interface");
   applet->priv->notify_id =
     gconf_client_notify_add (applet->priv->client,
diff -urN gnome-netstatus-2.26.0/src/netstatus-iface.c gnome-netstatus-2.26.0.new/src/netstatus-iface.c
--- gnome-netstatus-2.26.0/src/netstatus-iface.c	2009-03-08 06:25:52.000000000 +0100
+++ gnome-netstatus-2.26.0.new/src/netstatus-iface.c	2009-03-12 11:04:39.000000000 +0100
@@ -50,6 +50,7 @@
 
 #include "netstatus-sysdeps.h"
 #include "netstatus-enums.h"
+#include "netstatus-util.h"
 
 #define NETSTATUS_IFACE_POLL_DELAY       500  /* milliseconds between polls */
 #define NETSTATUS_IFACE_POLLS_IN_ERROR   10   /* no. of polls in error before increasing delay */
@@ -282,6 +283,71 @@
     }
 }
 
+gboolean
+netstatus_ifaces_monitor_start (void)
+{
+  static GSList  *iflist;
+  static guint    ifllen;
+  NetstatusIface *ifn;
+  GSList         *il;
+  GList          *iface_names, *l;
+
+  /*
+   * Add interface to our private interface list if it's not already on it.
+   */
+  iface_names = netstatus_list_interface_names (NULL);
+
+  if (ifllen < g_list_length(iface_names))
+    {
+      for (l = iface_names; l && l->data && strncmp(l->data, "lo", 2); l = l->next)
+        {
+          if (!g_slist_find_custom(iflist, l->data, (GCompareFunc) strcmp))
+            {
+              gchar *idata = g_malloc0(IF_NAMESIZE + 18);
+
+              strcpy(idata, l->data);
+              iflist = g_slist_append(iflist, idata);
+            }
+          g_free (l->data);
+        }
+    }
+
+  ifllen = g_list_length(iface_names);
+  g_list_free (iface_names);
+
+  /* Iterate thru all our private interface list and see if we've changed ip address */
+  for (il = iflist; il; il = g_slist_next(il))
+    {
+      gchar *oldipaddr = il->data + IF_NAMESIZE;
+      gchar *curaddr;
+
+      ifn = netstatus_iface_new(il->data);
+      netstatus_iface_get_inet4_details(ifn, &curaddr, NULL, NULL, NULL);
+
+      /*
+         *  If current IP address is NULL then interface is or has gone down, so
+         *  we don't notify user.
+         *
+         *  If however current IP changes state from down or some other IP to new
+         *  address then we notify user.
+       */
+
+      if (curaddr != NULL && (!*oldipaddr || strcmp(oldipaddr, curaddr)))
+        {
+          GString *str = g_string_new (NULL);
+          g_string_printf(str, _("%s is up with IP address %s"), ifn->priv->name, curaddr);
+          netstatus_message(str->str);
+          g_string_free(str, TRUE);
+        }
+
+      strcpy(oldipaddr, curaddr ? curaddr : "");
+      g_free(curaddr);
+      g_object_unref(ifn);
+    }
+
+  return TRUE;
+}
+
 NetstatusIface *
 netstatus_iface_new (const char *name)
 {
diff -urN gnome-netstatus-2.26.0/src/netstatus-iface.h gnome-netstatus-2.26.0.new/src/netstatus-iface.h
--- gnome-netstatus-2.26.0/src/netstatus-iface.h	2008-11-26 17:24:56.000000000 +0100
+++ gnome-netstatus-2.26.0.new/src/netstatus-iface.h	2009-03-12 11:04:57.000000000 +0100
@@ -85,6 +85,7 @@
 							      const char     **hw_name,
 							      char           **hw_addr);
 gboolean               netstatus_iface_get_is_loopback       (NetstatusIface  *iface);
+gboolean               netstatus_ifaces_monitor_start        (void);
 
 G_END_DECLS
 
diff -urN gnome-netstatus-2.26.0/src/netstatus-util.c gnome-netstatus-2.26.0.new/src/netstatus-util.c
--- gnome-netstatus-2.26.0/src/netstatus-util.c	2009-03-08 06:25:52.000000000 +0100
+++ gnome-netstatus-2.26.0.new/src/netstatus-util.c	2009-03-12 11:07:07.000000000 +0100
@@ -29,6 +29,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <string.h>
+#include <libnotify/notify.h>
 
 #ifdef G_ENABLE_DEBUG
 NetstatusDebugFlags _netstatus_debug_flags = NETSTATUS_DEBUG_NONE;
@@ -188,3 +189,16 @@
 
   return g_list_prepend (list, str);
 }
+
+void
+netstatus_message (char *message)
+{
+  NotifyNotification *n;
+
+  notify_init("Netstatus");
+  n = notify_notification_new (_("Network Status"), message, "gnome-netstatus-txrx", NULL);
+  notify_notification_set_timeout (n, 3000);
+  //gnome_sound_play("/usr/share/sounds/gnibbles/pop.wav");
+  notify_notification_show (n, NULL);
+  g_object_unref(G_OBJECT(n));
+}
diff -urN gnome-netstatus-2.26.0/src/netstatus-util.h gnome-netstatus-2.26.0.new/src/netstatus-util.h
--- gnome-netstatus-2.26.0/src/netstatus-util.h	2008-11-26 17:24:56.000000000 +0100
+++ gnome-netstatus-2.26.0.new/src/netstatus-util.h	2009-03-12 11:07:49.000000000 +0100
@@ -85,6 +85,8 @@
 							   gpointer        func_data,
 							   gpointer        alive_object);
 
+void                 netstatus_message                     (char *message);
+
 #ifdef G_ENABLE_DEBUG
 
 #include <stdio.h>
